### DISCUSSION TITLE
New to Codespaces. Is there a way to sync my settings across all codespaces?

### DISCUSSION THREAD

**Discussion by Wisdawms** [community/community#108027] [Codespaces] [answered]
Hello, friends. 
I'm very new to using GitHub Codespaces, it's a very cool feature so far.
Is there a way to sync my settings across all newly created codespaces? (like vscode profile, .gitconfig settings like aliases and such, and perhaps even the .zshrc file) across all newly created codespaces?
Thanks!

**Comment by spenserblack** [comment, marked as answer]
There is VS Code's builtin settings synchronization, but for more advanced features, you can use a `dotfiles` repository. [Here's the docs](https://docs.github.com/en/codespaces/setting-your-user-preferences/personalizing-github-codespaces-for-your-account#dotfiles) for more details, but basically it will do this:
1. clone `dotfiles` somewhere in your codespace (outside of the workspace, but you can find it by browsing the codespace's filesystem)
2. Check if a setup script (like `setup.sh`) exists
  - If it does does, run the script
  - If it doesn't, copy the dotfiles (files starting with `.`) from the `dotfiles` repository to your home directory

With a setup script, you can take advantage of environment variables for configuration that is specific to codespaces. For example:

```bash
if [ "$CODESPACES" == true ]; then
	# Codespace-specific setup
	git config --global core.editor code
else
	# Local-specific setup
	git config --global core.editor vim
fi
```

**Reply by Wisdawms** [reply]
Thank you! This was helpful indeed.
